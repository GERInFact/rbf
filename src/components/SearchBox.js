import React from "react";
import "./searchbox.css";

const SearchBox = (props) => {
  const { searchChange } = props;
  return (
    <div className="search-box">
      <input
        aria-label="Search field for robots"
        type="search"
        placeholder="search robots"
        className="search-box_input"
        onChange={searchChange}
      />
    </div>
  );
};

export default SearchBox;
