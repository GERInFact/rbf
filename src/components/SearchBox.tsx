import React from "react";
import "./searchbox.css";

interface ISearchBoxProps {
  searchChange(event: React.SyntheticEvent<HTMLInputElement>): void
}

const SearchBox = ({ searchChange }: ISearchBoxProps) : JSX.Element => {
  return (
    <div className="search-box">
      <input
        aria-label="Search field for robots"
        type="search"
        placeholder="search robots"
        className="search-box_input"
        onChange={searchChange}
      />
    </div>
  );
};

export default SearchBox;
