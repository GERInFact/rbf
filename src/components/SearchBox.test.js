import React from "react";
import renderer from "react-test-renderer";
import SearchBox from "./SearchBox";

it("expect to render SearchBox component", () => {
    const component = renderer.create(<SearchBox />);
    expect(component.toJSON()).toMatchSnapshot();
});