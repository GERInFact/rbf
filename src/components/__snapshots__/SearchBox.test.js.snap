// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render SearchBox component 1`] = `
<div
  className="search-box"
>
  <input
    aria-label="Search field for robots"
    className="search-box_input"
    placeholder="search robots"
    type="search"
  />
</div>
`;
